anacron/readtab.c

```diff
@@ -185,6 +206,11 @@ register_job(const char *periods, const char *delays,
     jr->next = NULL;
     Debug(("Read job - period=%d, delay=%d, ident=%s, command=%s",
 	   jr->period, jr->delay, jr->ident, jr->command));
+
+    return;
+error:
+    obstack_free(&tab_o, jr);
+    die_e("Cannot allocate memory.");
```

Comment:
Note that die_e exits the process and does not bother deallocating other data allocated by the process. So it is really strange to deallocate this particular thing.